der/derive/src/attributes.rs

```diff
@@ -184,7 +185,7 @@ impl FieldAttrs {
 
             let context_specific = match self.tag_mode {
                 TagMode::Explicit => {
-                    if self.extensible || self.optional {
+                    if self.extensible || self.optional || self.default.is_some() {
```

Comment:
Since `self.optional || self.default.is_some()` is getting checked in several places, it might make sense to make a binding or extract a method, e.g. `is_optional()`